File: src/main/kotlin/exam33/Exam33.java
package exam33;

public class Exam33 {
	class Person {
		private String name = ""; // 실제 저장되는 필드

		public Person(String name) {
			this.name = name;
		}

		public String getName() { // Getter (""일 경우 예외 발생)
			if (name.isEmpty()) {
				throw new IllegalStateException("이름이 설정되지 않았습니다.");
			}

			return name;
		}

		public void setName(String name) { // Setter ("" 방지)
			if (name.isEmpty()) {
				throw new IllegalArgumentException("이름은 비어 있을 수 없습니다.");
			}

			this.name = name;
		}
	}
}

File: src/main/kotlin/exam33/Exam33.kt
package exam33

class Person(
    var _name: String = ""
) {
    var name: String
        get() {
            if (_name.isEmpty()) throw IllegalStateException("이름이 설정되지 않았습니다.")
            return _name
        }
        set(value) {
            if (value.isEmpty()) throw IllegalArgumentException("이름은 비어 있을 수 없습니다.")
            _name = value
        }
}

fun main() {
    val p = Person()
    println(p.name) // java.lang.IllegalStateException: 이름이 설정되지 않았습니다.
    p.name = "" // java.lang.IllegalArgumentException: 이름은 비어 있을 수 없습니다.
}